src/script/lua_api/l_env.cpp

```diff
@@ -622,6 +906,136 @@ int ModApiEnvMod::l_add_entity(lua_State *L)
 	return 1;
 }
 
+// add_3dline(line_params) -> Lua3DLine or nil
+int ModApiEnvMod::l_add_3dline(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	LineParams line_params;
+
+	lua_getfield(L, 1, "type");
+	const char* type = luaL_checkstring(L, -1);
+
+	if (!strcmp(type, "flat"))
```

Comment:
pretty sure that C++ allows string comparison using `==`

Reply:
No, it compares only first chars. strcmp() compares whole strings.